lib/build-status.js

```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+const Joi = require('joi')
+
+const happyStatuses = ['passed', 'passing', 'success']
+
+const unhappyStatuses = ['error', 'failed', 'failing', 'unstable']
+
+const otherStatuses = [
+  'building',
+  'cancelled',
+  'expired',
+  'no tests',
+  'not built',
+  'not run',
+  'pending',
+  'processing',
+  'queued',
+  'running',
+  'scheduled',
+  'skipped',
+  'stopped',
+  'timeout',
+  'waiting',
+]
+
+const isBuildStatus = Joi.equal(
+  happyStatuses.concat(unhappyStatuses).concat(otherStatuses)
+)
+
+function renderBuildStatusBadge({ label, status }) {
```

Comment:
I like this idea!

In terms of reusability, should reformat `happyStatuses` to use our standard text? I know in a few places we do that kind of translation, i.e. display *passing* instead of *passed*. We could do the same thing for **failing** -> **failed**.

Reply:
Done!